src/git-source-provider.ts

```diff
@@ -0,0 +1,285 @@
+import * as core from '@actions/core';
+import * as coreCommand from '@actions/core/lib/command';
+import * as fs from 'fs';
+import * as fsHelper from './fs-helper';
+import * as gitCommandManager from './git-command-manager';
+import * as io from '@actions/io';
+import * as path from 'path';
+
+let authConfigKey = `http.https://github.com/.extraheader`;
+
+export async function getSource(
+    repositoryPath: string,
+    repositoryOwner: string,
+    repositoryName: string,
+    ref: string,
+    commit: string,
```

Comment:
I would make all of these arguments in the function into an interface and pass this as a single object to keep this simple.

Reply:
personally not a fan of that pattern. i find the indirection hinders readability (now i have to find the class to see what the params are)

Reply:
i added eslint/prettier to hopefully cut down on a lot of the style stuff

Reply:
actually the pattern may come in handy after i get jest setup... will see...

Reply:
done. i moved the args behind an interface. (lined-up with moving the input collection to a separate module for testability)